packages/astro/src/core/build/static-build.ts

```diff
@@ -186,6 +186,17 @@ async function ssrBuild(
 							const sanitizedName = name.split('.')[0];
 							return `chunks/${sanitizedName}_[hash].mjs`;
 						}
+						// Detect if the chunk name has as % sign that is not encoded.
+						// This is borrowed from Node core: https://github.com/nodejs/node/blob/3838b579e44bf0c2db43171c3ce0da51eb6b05d5/lib/internal/url.js#L1382-L1391
+						// We do this because you cannot import a module with this character in it.
+						for(let i = 0; i < name.length; i++) {
+							if(name[i] === '%') {
+								const third = name.codePointAt(i + 2)! | 0x20;
+								if (name[i + 1] !== '2' || third !== 102) {
+									return `chunks/${name.replace(/%/g, '_percent_')}_[hash].mjs`;
+								}
+							}
+						}
```

Comment:
I feel like it should be possible to import modules with `%` but it needs to be escaped before importing. Is it happening inside the Rollup-generated code, or are we importing it wrongly?

Reply:
It's the rollup generated code, it has this exactly like `import('../chunks/four%.hash.mjs')` and Node cannot handle that. I think this might actually be a Node.js bug; not being able to import valid filename seems like a bug to me. But any event, if we escape it in the filename then everything works.

Reply:
We talked offline and this is either a Node.js bug or a Rollup bug, so going with the workaround for now. I'll be filing a bug with Node.js to see what they think.